packages/fast-usdc/src/exos/advancer.js

```diff
@@ -148,11 +148,10 @@ export const prepareAdvancerKit = (
 
             const { borrowerFacet, poolAccount } = this.state;
             const { recipientAddress } = evidence.aux;
-            // throws if EUD is not found
-            const { EUD } = addressTools.getQueryParams(
-              recipientAddress,
-              EudParamShape,
-            );
+            const decoded = decodeAddressHook(recipientAddress);
+            mustMatch(decoded, AddressHookShape);
```

Comment:
I don't think we check that `cctpTxEvidence` contains the correct `settlementAccountAddress`. At one point I thought this was a concern of the EventFeed, but maybe this is the better place to do it.

To avoid increasing scope of this PR, I'll tackle this in a follow up